src/ferui-components/tree-view/tree-node-component.ts

```diff
@@ -176,9 +193,14 @@ export class FuiTreeNodeComponent<T> implements OnInit, OnDestroy, DoCheck {
 
   /**
    * Calculates the needed padding based on nodes level and if it has children
+   * 20 for the indentation pixels per level
+   * 10 pixels if has children to add equal 5 pixel margin around icon
+   * 30 pixels if no children to add margin from beginning of node text to beginning of left hand tree view
    */
   calculatePadding(): number {
-    return this.hasChildren ? this.level * 20 + 10 : this.level * 20 + 30;
+    // On first level, we check if any sibling nodes have children, if not we only add 10px padding and not 30 for icons
+    const padding = this.level === 0 ? (this.siblingHasChildren ? 30 : 10) : 30;
+    return this.hasChildren ? this.level * 20 + 10 : this.level * 20 + padding;
```

Comment:
Here again, the magical 20px... Use a constant.

Reply:
Done